core/src/client/mod.rs

```diff
@@ -132,12 +132,25 @@ pub trait TransportSenderT: MaybeSend + 'static {
 	/// Send.
 	async fn send(&mut self, msg: String) -> Result<(), Self::Error>;
 
+	/// Send ping frame (opcode of 0x9).
+	async fn send_ping(&mut self, data: &[u8]) -> Result<(), Self::Error>;
+
 	/// If the transport supports sending customized close messages.
 	async fn close(&mut self) -> Result<(), Self::Error> {
 		Ok(())
 	}
 }
 
+/// Message type received from the RPC server.
+/// It can either be plain `String` data, or a `Pong` reply to a previously submitted `Ping`.
+#[derive(Debug)]
+pub enum ReceivedMessage {
+	/// Incoming packet contains plain `String` data.
+   Data(String),
```

Comment:
Perhaps add `Binary(Vec<u8>)` as well or just wrap https://docs.rs/soketto/latest/soketto/data/enum.Data.html in `Data`